### ISSUE TITLE
[Vanilla Parity] Villager Enchanted Book Trades should Only have a Single Enchantment

### ISSUE THREAD

**Issue by WheelerShigley** [nulli0n/ExcellentEnchants-spigot#73]
In vanilla, a villager enchanted-book-trade has only one enchant and is at a random level (from minimum to maximum for the chosen enchant); from the [wiki](https://minecraft.fandom.com/wiki/Trading#cite_note-enchanted-book-11), "The enchantment is chosen randomly with equal chance of any enchantment type occurring (except for Soul Speed and Swift Sneak) and equal chance to get any level of the enchantment... The price in emeralds depends on the enchantment level and "treasure" status. The price is [maximum(``2+3*level,1)``,minimum(``6+13*level``,64)]... For treasure enchantments, the price is doubled, but still capped at 64 emeralds."

### @ su.nightexpress.excellentenchants.enchantment.util
```java
public static void singlePopulate(@NotNull ItemStack item) {
        //create map of vanilla enchants with their maximum level (possibly using EnchantRegistry.getRegistered() )
        //add custom enchants (with their maximum levels) to map

        //remove enchant from current enchanted book
        //select random item from map; select random level (1...entry_value); add selected enchant to book
}
```
### @ su.nightexpress.excellentenchants.enchantment.listener
```java
    // ---------------------------------------------------------------
    // Adding Enchants to Villagers
    // ---------------------------------------------------------------
    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
    public void onEnchantPopulateVillagerAcquire(VillagerAcquireTradeEvent e) {
        // ...
        if (!EnchantUtils.isEnchantable(result)) return;
        EnchantUtils.singlePopulate(result); // <------ Modification
        // ...
    }
```

**Comment by WheelerShigley** [comment]
Maybe I misunderstand how you're doing this, but when I set it to
```dot
    Villagers:
      Enabled: true
      Enchantments:
        Total_Maximum: 1
        Custom_Generation_Chance: 80.0
        Custom_Minimum: 0
        Custom_Maximum: 1
```
Villagers will always have a vanilla enchant on the book and 80% of the time, have an additional (single) custom enchant.

The desired functionality I'm trying to achieve is that  there exists only a single enchant on the enchanted book, vanilla or excellent_enchant both being enchants; is there some specific way in the config to replicate this functionality?

**Comment by nulli0n** [comment]
Could you join discord for a test .jar ?

**Comment by WheelerShigley** [comment]
It works well, but as it is now.
All levels are possible to get 5 emeralds for, which I've been fine with lol-
||The range is usually 3*level+2 to min(13*level+6,64).||

**Comment by nulli0n** [comment]
Fixed in v4.0.0 with Vanilla Distribution mode
